Cargo.toml

```diff
@@ -2,7 +2,7 @@
 name = "unicode-general-category"
 version = "1.0.0"
 authors = ["YesLogic Pty. Ltd. <info@yeslogic.com>"]
-edition = "2018"
+edition = "2024"
```

Comment:
We currently support back to Rust 1.40.0, this significantly increases the MSRV with no apparent benefits that I can see. I.e. the changes don't make use of any newly available features or behaviour that notably improve the code. I would suggest dropping f889998ff2a0e0ce7272fc09d881c10ae05b155f from the PR.